src/vcf/validator.cpp

```diff
@@ -181,7 +179,24 @@ namespace ebi
             << " compression";
     }
 
-    bool is_compressed_magic_num(const std::vector<char> &line)
+    void create_uncompressed_stream(std::istream & input,
+                                    const std::string & sourceName,
+                                    boost::iostreams::filtering_istream & uncompressed_input)
+    {
+        std::string file_ext = ebi::vcf::get_compressed_extension(sourceName);
+        std::set<std::string> readable_extentions = {BZ2,NO_EXT};
+
+        if (!readable_extentions.count(file_ext)) {
+            throw std::invalid_argument{"Input file should not be compressed"};
+        }
```

Comment:
I can see that this function will need to be modified to ask for the first line and check for the magic num as well.

It's ok until you add that.